chat/useCurrentChatUser.js

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from 'react';
+import { getStreamUser } from './utils';
+
+import { useCurrentUser } from 'hooks';
+
+/**
+ * Maintains user and token data needed to initialize chat.
+ * Automatically stays in sync with authentication state, and uses local
+ * state for the `chatUser` value so hook consumers can properly react
+ * to changes via `useEffect`, etc.
+ * @returns {Object}
+ */
+export default function useCurrentChatUser() {
+  const { currentUser } = useCurrentUser();
+  const [chatUser, setChatUser] = useState(null);
+
+  useEffect(() => {
+    if (currentUser && chatUser === null) {
+      setChatUser(getStreamUser(currentUser));
+    } else if (!currentUser && chatUser !== null) {
```

Comment:
Could this `else if` just be the default return or does this need to explicitly only `setChatUser` to null in this case?

Reply:
It's subtle and admittedly not the most legible, but, we only want to change the `chatUser` value if it's out of sync with the current user auth state.

I added a code comment to clarify this in c04d98a (and 4fd9643) since on second glance, it's a bit obtuse.